### ISSUE TITLE
Should hooks always point to actual executables

### ISSUE THREAD

**Issue by RenaudWasTaken** [cncf-tags/container-device-interface#6]
Hello!

As per @zvonkok's comment in https://github.com/container-orchestrated-devices/container-device-interface/pull/2 we should consider specifying that hooks must always point to actual executables and not symlinks.

> runc , for example, will not execute a runtime hook if it is a symlink. Should we mention this fact somewhere that hooks should be "actual" executables and not symlinks?

This might be something we could check in the go library or an alternative would be that we follow the symlink before merging it in the OCI spec.

Thoughts?
/cc @zvonkok

**Comment by adrianchiris** [comment]
I wonder if there is any reason for it? or just something that can be "fixed" in runc

**Comment by RenaudWasTaken** [comment]
Hello!

This topics was discussed in yesterday's meeting here are some of the notes:

TLDR:
- This is the case in runc to prevent security issues.
- We thought it made sense to start with a limit on pointing to executable and then expand if we see this as a blocker in the future.

More notes:
- For StartContainer it seems like you might need to follow symlinks.
- Should “runtime” hooks be in a root owned directory (e.g: /etc/cdi/bin)?
   - Concern about it being a “trashcan”
   - Vendors might want to install binaries in /usr/local/bin
- Would be nice to get more input / think about this more
